### PR TITLE
Filter box view component

### LINKED ISSUE #2087: Content Status Filter on Content Items List Suggestion
I think that the available selector options are close to what we have on the DB but they are not very useful to a content editor.
Maybe we should enable only: show published items, show drafted items and show owned by me items.
I don't see any value on the "latest" option on this list.  Let me know if I'm missing something.
And the options should be checkboxes to let the user mix and match.

What we have now:
show selector

And two possible alternatives:
show checkboxes

show multiselect selector

### LINKED ISSUE #2101: Odd Pager behavior in Content List Page
Repro:
1- List all content items (you should have more than 10).
2 - Go to page 2.
3 - Filter to show only content items of type landing page.

Expected: I should be on page 1. No pager shown.
Achieved: I'm on an empty page. Pager is shown.  Page 2 in on querystring. I can click "go to previous" on the pager and I go to the first page where I can see my landing page.

### LINKED ISSUE #2123: Content Filter, automatically apply Filter results, no need to manually click: "APPLY"
Content Filter, automatically apply Filter results, no need to manually click: "APPLY"

### PR DESCRIPTION
**Pull request by matiasmolleja**
I'm trying to improve the filter box. 
Fixing some issues and trying to make it reusable on other pages as suggested #2037 .
Added "Owned by me" feature. Added sort direction (asc, desc).
This PR is a first step. I just wanted to go step by step to make it easy to merge and see if the path I'm following is OK.

Fixes #2101
Fixes #2087
Fixes #2123

### CODE REVIEW

**Comment by sebastienros** [comment]
Did you follow the same UI for filters as VSTS?

**Comment by sebastienros** [comment]
Can you share a gif on this PR?

**Comment by matiasmolleja** [comment]
I did not worked yet on adapt to what VSTS do, but I will work on that next. 
Just wanted to separate the thing to its own component and fix a few things before.

Here is a gif:

File: src/OrchardCore.Modules/OrchardCore.Contents/Services/IQueryFromFilterBox.cs [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+using OrchardCore.ContentManagement.Records;
+using OrchardCore.Contents.ViewModels;
+using OrchardCore.DisplayManagement.ModelBinding;
+using OrchardCore.Navigation;
+using System.Threading.Tasks;
+using YesSql;
+
+namespace OrchardCore.Contents.Services
+{
+    public interface IQueryFromFilterBox
```
Comment (sebastienros) [inline src/OrchardCore.Modules/OrchardCore.Contents/Services/IQueryFromFilterBox.cs]: I don't think we need an interface, we can directly register and resolve the class, no need for an abstraction. Even more if this is not in the abstractions project.

Same for ISyerContentTypesProvider. Just merge the two classes into one.

Comment (sebastienros) [reply src/OrchardCore.Modules/OrchardCore.Contents/Services/IQueryFromFilterBox.cs]: You resolve the class itself, like you were doing before with the interface.

**Comment by sebastienros** [comment]
Can you explain why you are doing this refactoring though?
Why would we need to reuse this component? Even more if we know we need to work on the Content Tree at some point which will replace this. We pulled the filter because it was a simple solution that was good enough for the time begin.

**Comment by matiasmolleja** [comment]
It started working on it after this discussion: #2037

The idea was to made something that can be reused on other pages. I imagine for example a content item picker, or any other page where we need to show a list of content items. If we have a filterboxViewComponent and a ContentItemListViewComponent we could use them combined anywhere (although to be honest I did not fully validated this approach yet, and I'm not sure if it will be easy to reuse this thing) .

I wanted too to style the content list page to make it close to VSTS, as I recently did for the assets module. After pushing this PR I made pretty good progress on that and now in my local copy I have a nice looking list page.

When I started I saw a few issues and fixed them too ( #2087 , #2101 ).
If you think it is better I can stop working on this to avoid duplicating efforts with the content tree development.

**Comment by matiasmolleja** [comment]
I added the changes I mentioned about adding the new fluent-like style.
I added too a submit on change feature to avoid having to click on apply on each change (#2123)
There are still many thing that can be improved but maybe it's better to wait for the content-tree changes.
In the last commit I'm removing a couple of templates that are not needed after my changes. Let me know if it is ok to remove them.

content-list-fluent

**Comment by matiasmolleja** [comment]
Regarding the published-draft labels, I tried different options. I'm completely sure sure about what version do I prefer. We can change. Here are the options I have:

status-options

**Comment by sebastienros** [comment]
Looks nice, my own comments:
- I don't like the lower-casing of the headers
- You could use a single icon (in the green and blue buttons) and the text as tootlip. Header would just be "Status"
- I prefer to see the `one day ago` text instead of full dates, which can go in tooltips instead
- You just need to show Created. The modified date can be in the draft icon tooltip and the published on the published icon.

You are the designated owner of the Content Tree now ;) Congrats !

**Comment by matiasmolleja** [comment]
After trying the last changes, I think it's weird having a selector to sort by modified date, or published date, and not having the corresponding columns on the table. The user will switch for example to modified date, ascending, and he won't have a way to verify that the sorting has been applied (other than start hovering over the icons in sequence).
If the reason to hide the columns is unclutter the view, maybe it's better to let the user show/hide colums (as VSTS do). That can be quite easy to do on the client.

**Comment by sebastienros** [comment]
I am closing this PR because we need some better reasons to have this component first. And it's old and not active. I am not deleting the branch for now as I assume you would not like to lose the code.

**Pull request closed without merging**